crates/sui-core/src/authority/authority_store.rs

```diff
@@ -1155,6 +1157,22 @@ impl<S: Eq + Debug + Serialize + for<'de> Deserialize<'de>> SuiDataStore<S> {
             .contains_key(digest)?)
     }
 
+    pub async fn consensus_message_processed_notify(
+        &self,
+        digest: &TransactionDigest,
+    ) -> Result<bool, SuiError> {
+        let registration = self.consensus_notify_read.register_one(digest);
+        if self
+            .epoch_tables()
+            .consensus_message_processed
+            .contains_key(digest)?
+        {
+            return Ok(true);
```

Comment:
Why return bool instead of ()? And what is the disadvantage of using `notify_read()` here?

Reply:
Yeah, there is no reason to return bool - changed to `()`

>And what is the disadvantage of using notify_read() here?

Can you elaborate on this?

Reply:
I misread the code. `notify_read()` is implemented for `EffectsNotifyRead` trait and this is an implementation similar to `notify_read()` for consensus message processed.